scripts/paypal/payment-reconciliator.ts

```diff
@@ -100,47 +102,85 @@ const markOrderAsError = async order => {
   await order.update({ status: OrderStatus.ERROR });
 };
 
-const findOrdersWithErroneousStatus = async (hostSlug, fix = false) => {
-  if (!hostSlug) {
-    return exitWithUsage();
+const findOrdersWithErroneousStatus = async (hostSlugs: string[], fix = false) => {
+  for (const hostSlug of hostSlugs) {
+    console.log(`\nChecking host ${hostSlug} for erroneous order statuses...`);
+    const host = await models.Collective.findBySlug(hostSlug);
+    const orderIterator = getPaypalPaymentOrdersIterator(host, { status: 'PAID' });
+    let orderItem = await orderIterator.next();
+    let hasError = false;
+    while (!orderItem.done) {
+      const isValid = await checkOrder(host, orderItem.value);
+      hasError = hasError || !isValid;
+
+      if (fix && !isValid && !parseToBoolean(process.env.DRY)) {
+        await markOrderAsError(orderItem.value);
+      }
+
+      orderItem = await orderIterator.next();
+    }
+
+    if (!hasError) {
+      logger.info(`No error found for PayPal orders for host ${hostSlug}`);
+    }
   }
+};
 
-  const host = await models.Collective.findBySlug(hostSlug);
-  const orderIterator = getPaypalPaymentOrdersIterator(host, { status: 'PAID' });
-  let orderItem = await orderIterator.next();
-  let hasError = false;
-  while (!orderItem.done) {
-    const isValid = await checkOrder(host, orderItem.value);
-    hasError = hasError || !isValid;
+/**
+ * See https://developer.paypal.com/docs/api/transaction-search/v1/#transactions-get-query-parameters
+ */
+const getRefundedTransactionsFromPaypal = async (host, startDate, endDate) => {
+  const urlParams = new URLSearchParams();
+  urlParams.append('fields', 'all');
+  urlParams.append('page_size', '100');
+  urlParams.append('page', '1');
+  urlParams.append('transaction_status', 'V');
+  urlParams.append('start_date', startDate.toISOString());
+  urlParams.append('end_date', endDate.toISOString());
+  const apiUrl = `reporting/transactions?${urlParams.toString()}`;
+  const response = await paypalRequest(apiUrl, null, host, 'GET');
+  // TODO: Handle pagination
+  return response['transaction_details'];
+};
 
-    if (fix && !isValid && !parseToBoolean(process.env.DRY)) {
-      await markOrderAsError(orderItem.value);
+const findRefundedContributions = async (hostSlugs: string[]) => {
+  for (const hostSlug of hostSlugs) {
+    console.log(`\nChecking host ${hostSlug} for refunded contributions not marked as such in our ledger...`);
```

Comment:
Aren't we just listing refunded transactions?

Reply:
For now, yes. Sorry, I should have indicated that this script was not complete in the PR description